network/config.go

```diff
@@ -128,7 +128,7 @@ type Config struct {
 	// TLSKey is this node's TLS key that is used to sign IPs.
 	TLSKey crypto.Signer `json:"-"`
 	// BLSKey is this node's BLS key that is used to sign IPs.
-	BLSKey *bls.SecretKey `json:"-"`
+	BLSKey bls.Signer `json:"-"`
```

Comment:
I assume this means the `BLSKey` property gets serialized as the string, `"-"`?

Reply:
IIRC this actually omits it from serialization, i.e it's not serialized or deserialized. This is done because we use the `Config` struct to also hold fields and not just config values.... so our hack is to ignore serialization for non-config values (which is weird IMO).

Reply:
Makes sense. 

Yeah, I generally prefer to split up raw config data vs actual dependent services/structs, but 🤷. I've seen this pattern before too.

Reply:
one day...